intra/protect/xdial.go

```diff
@@ -201,3 +210,24 @@ func (d *RDial) AcceptTCP(network string, local string) (*net.TCPListener, error
 		return nil, errNoTCPMux
 	}
 }
+
+func setKeepAliveConfig(tc *net.TCPConn, keepIdle, keepCnt, keepIntvl int){
```

Comment:
minor: could move this fn and other constants to a new file `intra/protect/sockopt.go`?